### DISCUSSION TITLE
createFormOptions leading to infinite loop…

### DISCUSSION THREAD

**Discussion by gregoryloichot** [filamentphp/filament#14737] [Help] [answered]
### Package

Panel builder

### Package Version

v3.2

### How can we help you?

Hello, 

In the table of the resource Human, I have a field called "parent" and listing "Human". I wanted to add a `createFormOptions` to this field, but then we have the following infinite loop situation:

```php
public static function form(Form $form): Form
    {
        return $form
            ->schema(self::getFormSchema());
    }
 public static function getFormSchema(): array
    {
        return [
                    …
                    Forms\Components\Select::make('parent')
                        ->relationship(titleAttribute: 'name')
                        ->native(false)
                        ->preload()
                        ->searchable()
                        ->createOptionForm(self::getFormSchema()),
                       …
         ];
}
```

Is there any solution ? I have to create the `getFormSchema` because I reuse this form in many other forms.

**Comment by ModestasV** [comment, marked as answer]
This is expected since you are recursively calling this.

`getFormSchema()` contains your Select that calls `getFormSchema()` - this causes an infinite loop.

To avoid this, add a property to the `getFormSchema()` to remove the select. For example, `getFormSchema($withSelect = true)` and call it with that parameter set to `false` in your Select option form.
